src/collective.jl

```diff
@@ -167,6 +167,38 @@ Scatter(sendbuf, T, comm; root::Integer=Cint(0)) =
 Scatter(sendbuf, ::Type{T}, root::Integer, comm::Comm) where {T} =
     Scatter!(sendbuf, Ref{T}(), root, comm)[]
 
+"""
+    Scatter(arr::AbstractVector, comm::Comm; root::Cint=0)
+
+Splits a 1D array `arr` with elements of the same type in the `root` process into `nprocs=Comm_size(comm)` smaller 1D arrays. 
+The array `arr` is splitted in rank order, and the number of elements `n=length(arr)` can be not divisible by `nprocs`.
+Each process with the rank `j` returns a smaller array with the number of elements 
+`j < rem(n,nprocs) ? div(n,nprocs) + 1 : div(n,nprocs)`.
+"""
+function Scatter(arr::Union{Nothing, AbstractVector}, comm; root=Cint(0))
+  rank = MPI.Comm_rank(comm)
+  nprocs = MPI.Comm_size(comm)
+
+  arr_len = 0
+  elm_t = nothing
+  if rank == root
+    arr_len = length(arr)
+    elm_t = eltype(arr)
+  end
+  arr_len = MPI.Bcast(arr_len, root, comm)
+  elm_t = MPI.bcast(elm_t, root, comm)
```

Comment:
Why is one of these `Bcast` and the other `bcast`?

Reply:
If I remember correctly, `elm_t` is not a `isbits` type and can only be broadcasted with `bcast`. My idea was to use `Bcast` whenever possible, as its overhead should be less than `bcast`(?). Of course, using `bcast` for both `elm_t` and `arr_len` should work. I think it is a matter of style. What do you think?